Struct example doesn't work (maybe just for me?)

Hey people, I was trying to run the [struct example](https://docs.rs/mockall/0.5.2/mockall/#mocking-structs) but it didn't work. Some of the errors were intuitive to fix, but I didn't get too far. Can you guide in making it work? And nice work by the way.

The whole example:

```rust
use mockall::*;
use crate::thing::Thing;

mod thing {
    pub struct Thing{}
    #[automock]
    impl Thing {
        pub fn foo(&self) -> u32 {
            return 32;
        }
    }
}

cfg_if! {
    if #[cfg(test)] {
        use self::thing::MockThing as Thing;
    } else {
        use self::thing::Thing;
    }
}

fn do_stuff(thing: &Thing) -> u32 {
    thing.foo()
}

#[cfg(test)]
mod t {
    use super::*;

    #[test]
    fn test_foo() {
        let mut mock = Thing::default();
        mock.expect_foo().returning(|| 42);
        do_stuff(&mock);
    }
}

```

The error with rust 1.36:

```sh
error[E0658]: The attribute `automock` is currently unknown to the compiler and may have meaning added to it in the future
 --> src/main.rs:6:7
  |
6 |     #[automock]
  |       ^^^^^^^^
  |
  = note: for more information, see https://github.com/rust-lang/rust/issues/29642

error: cannot find macro `cfg_if!` in this scope
  --> src/main.rs:14:1
   |
14 | cfg_if! {
   | ^^^^^^

warning: unused import: `mockall::*`
 --> src/main.rs:1:5
  |
1 | use mockall::*;
  |     ^^^^^^^^^^
  |
  = note: #[warn(unused_imports)] on by default

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0658`.
error: Could not compile `mockall_example`.

To learn more, run the command again with --verbose.
```

The error with rust 1.39:

```sh
error: cannot find macro `cfg_if` in this scope
  --> src/main.rs:14:1
   |
14 | cfg_if! {
   | ^^^^^^

error: cannot find attribute `automock` in this scope
 --> src/main.rs:6:7
  |
6 |     #[automock]
  |       ^^^^^^^^

warning: unused import: `mockall::*`
 --> src/main.rs:1:5
  |
1 | use mockall::*;
  |     ^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error: aborting due to 2 previous errors

error: could not compile `mockall_example`.

To learn more, run the command again with --verbose.
```

and not necessarily related: I'm doing an evaluation of mocking libraries specific for structs and you're my last hope haha. I tested others with conditional compilation but yours seems more promising and complete. And I'm wondering how it works when the struct is composed of more complex structs. I would love to help more in the crate.

--- a/mockall/src/lib.rs
+++ b/mockall/src/lib.rs
@@ -579,10 +579,9 @@
 //! [`#[automock]`](attr.automock.html)
 //! works for structs that have a single `impl` block:
 //! ```no_run
-//! # use mockall::*;
 //! # use cfg_if::cfg_if;
 //! mod thing {
-//!     # use mockall::automock;
+//!     use mockall::automock;
 //!     pub struct Thing{}
 //!     #[automock]
 //!     impl Thing {
@@ -916,7 +915,7 @@
 //! # use mockall::*;
 //! # use cfg_if::cfg_if;
 //! mod ffi {
-//!     # use mockall::automock;
+//!     use mockall::automock;
 //!     #[automock(mod mock;)]
 //!     extern "C" {
 //!         pub fn foo(x: u32) -> i64;
@@ -964,7 +963,7 @@
 //! # use mockall::*;
 //! # use cfg_if::cfg_if;
 //! mod outer {
-//!     # use mockall::automock;
+//!     use mockall::automock;
 //!     #[automock()]
 //!     pub(super) mod inner {
 //!         pub fn bar(x: u32) -> i64 {

